File: src/frogger/screen/frame/GameFrame.java
package frogger.screen.frame;

import frogger.screen.frame.elements.map.Map;
import javafx.fxml.FXMLLoader;
import javafx.fxml.Initializable;
import javafx.scene.layout.AnchorPane;

import java.io.IOException;
import java.net.URL;
import java.util.ResourceBundle;

/**
 * @author Lucas Hagen.
 */
public class GameFrame extends AnchorPane implements Initializable {

    public GameFrame() throws IOException {
        FXMLLoader fxmlLoader = new FXMLLoader(getClass().getResource("GameFrame.fxml"));

        fxmlLoader.setRoot(this);
        fxmlLoader.setController(this);
        fxmlLoader.load();
    }

    public void generateMap() throws IOException {
        getChildren().add(new Map());
    }

    @Override
    public void initialize(URL location, ResourceBundle resources) {
        try {
            generateMap();
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: src/frogger/screen/frame/GameFrame.scala
package frogger.screen.frame

import java.net.URL
import java.util.ResourceBundle

import frogger.screen.frame.elements.map.Map
import javafx.fxml.{FXMLLoader, Initializable}
import javafx.scene.layout.AnchorPane

class GameFrame extends AnchorPane with Initializable {

  val fxmlLoader: FXMLLoader = new FXMLLoader(
    getClass.getResource("GameFrame.fxml"))

  fxmlLoader.setRoot(this)

  fxmlLoader.setController(this)

  fxmlLoader.load()

  private def generateMap(): Unit = {
    getChildren.add(new Map())
  }

  override def initialize(location: URL, resources: ResourceBundle): Unit = {
    try generateMap()
    catch {
      case e: Exception => e.printStackTrace()

    }
  }

}